Dev

DaimonV15 opened the pull request.

DaimonV15 committed:
Merge remote-tracking branch 'upstream/dev' into dev

DaimonV15 committed:
Task03 Dimitrov

распарсил json, суммммировал баланс по фильтру и вывод в файл.
Красивую табл. не успел %((

xthebat commented on dimitrov/task03/src/test/kotlin/ru/bmstu/iu4/task03/FileReadFunctionKtTest.kt:
```diff
@@ -0,0 +1,17 @@
+
+    @Test
+    fun sumBalanceTest() {
+        val loadFile = "C:/Users/dimit/OneDrive/Документы/generated.json"
```
У меня этот тест не запустится

xthebat commented on dimitrov/task03/src/main/kotlin/ru/bmstu/iu4/task03/FileReadFunction.kt:
```diff
@@ -0,0 +1,66 @@
+        "|name|", "|gender|", "|company|", "|email|", "|phone|", "|address|", "|about|", "|registered|", "|latitude|", "|longitude|",
+        "|tags|", "|friends|", "|greeting|", "|favoriteFruit|\n")*/
+
+    for (i in 0..global.lastIndex) {
```
Должно быть разделено на функции, а также сделано с помощью использования функций работы с коллекциями в функциональном стиле.

xthebat reviewed (changes requested):
- В названии Pull-request пишите какой таск вы сдаете
- Тесты должны проходить на любых компах не зависимо от окружения. Для это используйте либо ресуры. Либо хардкод в текст теста.
- Задание предполагало использование функционального стиля.

DaimonV15 committed:
task04

1. Выравнивание по центру
2. Добавил возможность "очищения таблицы" от двойных границ
3. Перенос текста внутри ячейки
4. Выравнивание по вертикали и горизонтали
5. Постарался сделать тесты %))

xthebat commented on dimitrov/task03/src/main/kotlin/ru/bmstu/iu4/task03/Starter.kt:
```diff
@@ -0,0 +1,14 @@
+    @JvmStatic
+    fun main(args: Array<String>) {
+        log.info { "Start:" }
+        log.info { outputFile("C:/Users/dimit/OneDrive/Документы/generated.json") }
```
не используйте абсолютные пути

DaimonV15 committed:
Task 05 + change Task 04

В 4ом задании поправил почти все что было помечено на гите
В 5ом задании реализовал функции отрисовки точки, линий, квадрата и круга
